plugins/inputs/cisco_telemetry_mdt_netconf/cisco_telemetry_mdt_netconf.go

```diff
@@ -0,0 +1,930 @@
+func (dsrs *dialinSubscriptionRequestsService) createRequests(
+	ctx context.Context, c *CiscoTelemetryNETCONF) error {
+	var err error
+
+	if len(dsrs.Subscriptions) > 0 {
+		// Initialization of subscriptions
+		subscriptions := make([]*netconfSubscriptionRequest, len(dsrs.Subscriptions))
+		for i, subscription := range dsrs.Subscriptions {
+			if subscription.Period != 0 {
+				if time.Duration(subscription.Period) >= 0 {
+					// Check type of subscription
+					switch subscription.UpdateTrigger {
+					case "periodic":
+						if time.Duration(subscription.Period) > 0 {
+							subscriptions[i] = &netconfSubscriptionRequest{
+								YangPush:    "urn:ietf:params:xml:ns:yang:ietf-yang-push",
+								Stream:      "yp:yang-push",
+								XPathFilter: subscription.XPathFilter,
+								Period: uint64(time.Duration(subscription.Period).Nanoseconds() /
+									(int64(time.Millisecond) * 10)),
+							}
+						} else {
+							err = fmt.Errorf(
+								"failed to create telemetry subscription: "+
+									"time period for subscription %d has to be "+
+									"strictly positive but is %d",
+								i+1,
+								time.Duration(subscription.Period),
+							)
+							c.acc.AddError(err)
+							continue
+						}
+					case "on-change":
+						p := new(uint64)
+						*p = uint64(time.Duration(subscription.Period).Nanoseconds() /
+							(int64(time.Millisecond) * 10))
+						// on-change
+						subscriptions[i] = &netconfSubscriptionRequest{
+							YangPush:        "urn:ietf:params:xml:ns:yang:ietf-yang-push",
+							Stream:          "yp:yang-push",
+							XPathFilter:     subscription.XPathFilter,
+							DampeningPeriod: p,
+						}
+					default:
+						err = fmt.Errorf(
+							"failed to create telemetry subscription: " +
+								"bad / missing field update_trigger " +
+								"(options are: periodic, on-change)",
+						)
+						c.acc.AddError(err)
+						continue
+					}
+
+					// Create dynamic subscription over NETCONF
+					c.Log.Debugf(
+						"Establishing subscription %s %s %s tags(s)=%v...",
+						subscription.XPathFilter, subscription.UpdateTrigger,
+						time.Duration(subscription.Period), subscription.Tags,
+					)
+
+					dsrs.Mutex.Lock()
+					err = dsrs.Session.CallSimple(subscriptions[i])
+					dsrs.Mutex.Unlock()
+
+					if err != nil {
+						c.reportError(ctx, err, "create telemetry subscription")
+					} else {
+						c.Log.Debugf(
+							"Established subscription",
+						)
+						// Announce state:created to the next stage of the pipeline
+						dsrs.setting.state = created
+					}
+				} else {
+					err = fmt.Errorf(
+						"failed to create telemetry subscription: "+
+							"time period for subscription %d has to be positive but is %d",
+						i+1,
+						time.Duration(subscription.Period),
+					)
+					c.acc.AddError(err)
+				}
+			} else {
+				err = fmt.Errorf(
+					"failed to create telemetry subscription: "+
+						"missing field in subscription %d: period",
+					i+1,
+				)
```

Comment:
Please also move this check to `Init()` to error out early and once as this error is not recoverable.